sklearn/model_selection/_split.py

```diff
@@ -51,12 +53,30 @@
 ]
 
 
-class BaseCrossValidator(metaclass=ABCMeta):
+class GroupsComsumerMixin(_MetadataRequester):
+    """A Mixin to ``groups`` by default.
+
+    This Mixin makes the object to request ``groups`` by default as
+    ``REQUESTED``, instead of the ``ERROR_IF_PASSED`` default value.
+
+    .. versionadded:: 1.1
+    """
+
+    __metadata_request__split = {"groups": RequestType.REQUESTED}
```

Comment:
Since this mixin does not really do too much, can we have the subclass set `__metadata_request__split` directly?

Reply:
For me it comes down to what the "developer API" should be. Should developers depend on  using a mixin or learn about the metadata routing API and set `__metadata_request__split` directly?

For the group splitters, I think:

```python
__metadata_request__split = {"groups": RequestType.REQUESTED}
```

has clear semantics and does not require a comment.

Reply:
Actually, I am okay with this for now since `sklearn/model_selection/_split.py` is private.